Add support for GNOME 45

rtpt-lucasvater opened the pull request:
I added support for GNOME 45 and made some changes in how the windows are moved to work better on Wayland with a Multi-Monitor setup. This should also work for X-based systems. Feel free to only take the changes you need for GNOME 45 support, if you don't want to change the window moving logic.

However, I didn't test if the changes are backward compatible with older GNOME versions. This might need some testing.

Closes #9

Commit:
Add support for Gnome 45

thomasaull commented:
Thanks for your work on this @rtpt-lucasvater! I can confirm it seems to work like it should on Fedora 39 with X11

troyready committed:
drop support for pre-v45

New import system breaks compatibility with older GNOME versions: https://blogs.gnome.org/shell-dev/2023/09/02/extensions-in-gnome-45/

troyready commented:
Thanks a ton for this @rtpt-lucasvater !
